aiida/tools/visualization/graph.py

```diff
@@ -405,69 +400,48 @@ def __init__(
         self._node_id_type = node_id_type
         self._backend = backend or get_manager().get_profile_storage()
 
-        self._ignore_node_style = _OVERRIDE_STYLES_DICT['ignore_node']
-        self._origin_node_style = _OVERRIDE_STYLES_DICT['origin_node']
+        self._ignore_node_style = _OVERRIDE_STYLES_DICT['ignore_node'].copy()
```

Comment:
Maybe instead of putting the responsibility on the caller of passing a clone to `add_node` just have `add_node` create a `copy` (maybe a `deepcopy` in case in the future the keys of `_OVERRIDE_STYLES_DICT` become nested). This will reduce the risk of a user forgetting to create a copy when calling `add_node` manually. Example is line 823 where the dictionary passed in is not copied and will be modified.

Reply:
I just converted `_OVERRIDE_STYLES_DICT` to a function, so its no longer mutable